python/src/uagents/agent.py

```diff
@@ -241,12 +337,18 @@ def __init__(
         self._replies: Dict[str, Dict[str, Type[Model]]] = {}
         self._queries: Dict[str, asyncio.Future] = {}
         self._dispatcher = dispatcher
+        self._dispenser = dispenser
         self._message_queue = asyncio.Queue()
         self._on_startup = []
         self._on_shutdown = []
         self._test = test
         self._version = version or "0.1.0"
 
+        # configure the dispenser
+        self._dispenser.configure(
+            resolver=self._resolver,
+        )
```

Comment:
Since the dispenser depends on a resolver, unlike the dispatcher, should we simply create it here instead of modifying this global which would affect all other usage of it?

Reply:
I was thinking the exact same thing but somehow still decided to have it globally available. Creating it here would mean to pass it down to the Context and my initial goal was to keep the context as small as possible.

If there are no more big changes it would make sense to create the dispenser here (since it is intended as a 1 per one agent basis anyway)